File: Golang/Medium/2348_Number_of_Zero_Filled_Subarrays/solution.go
package numberofzerofilledsubarrays

func zeroFilledSubarray(nums []int) int64 {
	count := 0
	totalCount := int64(0)

	for _, num := range nums {
		if num == 0 {
			count += 1
			totalCount += int64(count)
		} else {
			count = 0
		}
	}
	return totalCount
}

File: Python/Medium/2348_Number_of_Zero_Filled_Subarrays/solution.py
class Solution:
    def zeroFilledSubarray(self, nums: list[int]) -> int:
        count = 0
        totalCount = 0

        for num in nums:
            if num == 0:
                count += 1
                totalCount += count
            else:
                count = 0
        return totalCount